homeassistant/components/fan/dyson.py

```diff
@@ -10,62 +10,153 @@
+SERVICE_SET_TIMER = 'dyson_set_timer'
+SERVICE_SET_DYSON_SPEED = 'dyson_set_speed'
 
 DYSON_SET_NIGHT_MODE_SCHEMA = vol.Schema({
-    vol.Required(CONF_ENTITY_ID): cv.entity_id,
-    vol.Required(CONF_NIGHT_MODE): cv.boolean,
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_NIGHT_MODE): cv.boolean,
+})
+
+DYSON_SET_AUTO_MODE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_AUTO_MODE): cv.boolean,
+})
+
+DYSON_SET_ANGLE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_ANGLE_LOW): cv.positive_int,
+    vol.Required(ATTR_ANGLE_HIGH): cv.positive_int
+})
+
+DYSON_SET_FLOW_DIRECTION_FRONT_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_FLOW_DIRECTION_FRONT): cv.boolean
+})
+
+DYSON_SET_TIMER_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_TIMER): cv.positive_int
+})
+
+DYSON_SET_DYSON_SPEED_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+    vol.Required(ATTR_DYSON_SPEED): cv.positive_int
 })
 
 
 def setup_platform(hass, config, add_entities, discovery_info=None):
     """Set up the Dyson fan components."""
-    from libpurecoollink.dyson_pure_cool_link import DysonPureCoolLink
+    from libpurecool.dyson_pure_cool_link import DysonPureCoolLink
+    from libpurecool.dyson_pure_cool import DysonPureCool
+
+    if discovery_info is None:
+        return
 
     _LOGGER.debug("Creating new Dyson fans")
     if DYSON_FAN_DEVICES not in hass.data:
         hass.data[DYSON_FAN_DEVICES] = []
 
     # Get Dyson Devices from parent component
-    for device in [d for d in hass.data[DYSON_DEVICES] if
-                   isinstance(d, DysonPureCoolLink)]:
-        dyson_entity = DysonPureCoolLinkDevice(hass, device)
-        hass.data[DYSON_FAN_DEVICES].append(dyson_entity)
+    has_purecool_devices = False
+    device_names = (device.name for device in hass.data[DYSON_FAN_DEVICES])
+    for device in hass.data[DYSON_DEVICES]:
+        if device.name not in device_names:
+            if isinstance(device, DysonPureCool):
+                has_purecool_devices = True
+                dyson_entity = DysonPureCoolDevice(device)
+                hass.data[DYSON_FAN_DEVICES].append(dyson_entity)
+            elif isinstance(device, DysonPureCoolLink):
+                dyson_entity = DysonPureCoolLinkDevice(hass, device)
+                hass.data[DYSON_FAN_DEVICES].append(dyson_entity)
 
     add_entities(hass.data[DYSON_FAN_DEVICES])
 
     def service_handle(service):
         """Handle the Dyson services."""
-        entity_id = service.data.get(CONF_ENTITY_ID)
-        night_mode = service.data.get(CONF_NIGHT_MODE)
-        fan_device = next([fan for fan in hass.data[DYSON_FAN_DEVICES] if
-                           fan.entity_id == entity_id].__iter__(), None)
+        entity_id = service.data.get(ATTR_ENTITY_ID)
```

Comment:
```suggestion
        entity_id = service.data[ATTR_ENTITY_ID]
```
Please use `dict[key]` for required keys and for keys that have default values in the validation schema.